docs/site/reference/formatter-spec.md

```diff
@@ -60,3 +60,54 @@ outputs.
 ### 4. Reliable
 
 We expect the formatter to be reliable and not break the semantics of the formatted files.
+
+## Stdin Specification
+
+Formatters **MAY** also implement the [Stdin Specification](#stdin-specification), which allows
+formatting "virtual files" passed via stdin.
+
+A formatter **MUST** implement the [Stdin Specification](#stdin-specification) if its formatting behavior
+can depend on the name of the file being formatted.
+
+### Rules
+
+In order for the formatter to comply with this spec, it **MUST** satisfy the following:
+
+#### 1. `--stdin` flag
+
+The formatter's CLI must be of the form:
+
+```
+<command> [options] [--stdin] [...<files>]
+```
+
+Where:
+
+- `<command>` is the name of the formatting tool.
+- `[options]` is any number of flags and options that the formatter accepts.
+- `--stdin` is an optional flag that puts the formatter in "stdin mode". In
+  stdin mode, the formatter reads file contents from stdin rather than the
+  filesystem.
+- `[...<files>]` is one or more files given to the formatter for processing. If
+  `--stdin` is specified, then exactly 1 must be present, and is treated as a
+  filename.
+
+Example:
+
+```
+$ echo "{}" | nixfmt --stdin path/to/file.nix
```

Comment:
This is a bad example, as [nixfmt doesn't actually behave this way yet](https://github.com/NixOS/nixfmt/issues/305).

Do folks know if there's an existing formatter (besides treefmt itself) that implements this spec?

Reply:
Hmmm, then I can't say I know of any formatter that actually conforms to this yet. Sounds like `ormolu` might be close?

Reply:
Bummer. I'm going to leave this as is, then. Hopefully we can change nixfmt to honor this spec before we merge this.

Reply:
Discussed at today's Nix Formatting team meeting: we like `--stdin-filepath=<path>` for 2 reasons:

1. This is path, not a filename.
2. It's more obvious that this expects exactly 1 path, rather than `[files...]`

Reply:
I've updated the spec to use `--stdin-filepath`